pallets/parachain-staking/src/benchmarks.rs

```diff
@@ -33,14 +105,458 @@ benchmarks! {
+			}
+			0u32
+		};
+		col_nom_count.insert(collators[0].clone(), nominators.len() as u32);
+		// FILL remaining nominations
+		if remaining_nominations > 0 {
+			for (col, n_count) in col_nom_count.iter_mut() {
+				if n_count < &mut (nominators.len() as u32) {
+					// assumes nominators.len() <= MaxNominatorsPerCollator
+					let mut open_spots = nominators.len() as u32 - *n_count;
+					while open_spots > 0 && remaining_nominations > 0 {
+						let caller = nominators[open_spots as usize - 1usize].clone();
+						if let Ok(_) = Pallet::<T>::nominate(RawOrigin::Signed(
+							caller.clone()).into(),
+							col.clone(),
+							<<T as Config>::MinNominatorStk as Get<BalanceOf<T>>>::get(),
+							*n_count,
+							collators.len() as u32, // overestimate
+						) {
+							*n_count += 1;
+							remaining_nominations -= 1;
+						}
+						open_spots -= 1;
+					}
+				}
+				if remaining_nominations == 0 {
+					break;
+				}
+			}
+		}
+		// STORE starting balances for all nominators
+		let nominator_starting_balances: Vec<(
+			T::AccountId,
+			<<T as Config>::Currency as Currency<T::AccountId>>::Balance
+		)> = nominators.iter().map(|x| (x.clone(), T::Currency::free_balance(&x))).collect();
+		// PREPARE RUN_TO_BLOCK LOOP
+		let before_running_round_index = Pallet::<T>::round().current;
+		let round_length: T::BlockNumber = Pallet::<T>::round().length.into();
+		let reward_delay = <<T as Config>::BondDuration as Get<u32>>::get() + 2u32;
+		let mut now = <frame_system::Pallet<T>>::block_number();
+		let mut counter = 0usize;
+		let end = Pallet::<T>::round().first + (round_length * reward_delay.into());
+		// SET collators as authors for blocks from now - end
+		while now < end {
+			let author = collators[counter % collators.len()].clone();
+			Pallet::<T>::note_author(author);
+			<frame_system::Pallet<T>>::on_finalize(<frame_system::Pallet<T>>::block_number());
+			<frame_system::Pallet<T>>::set_block_number(
+				<frame_system::Pallet<T>>::block_number() + 1u32.into()
+			);
+			<frame_system::Pallet<T>>::on_initialize(<frame_system::Pallet<T>>::block_number());
+			Pallet::<T>::on_initialize(<frame_system::Pallet<T>>::block_number());
+			now += 1u32.into();
+			counter += 1usize;
+		}
+		Pallet::<T>::note_author(collators[counter % collators.len()].clone());
+		<frame_system::Pallet<T>>::on_finalize(<frame_system::Pallet<T>>::block_number());
+		<frame_system::Pallet<T>>::set_block_number(
+			<frame_system::Pallet<T>>::block_number() + 1u32.into()
+		);
+		<frame_system::Pallet<T>>::on_initialize(<frame_system::Pallet<T>>::block_number());
+	}: { Pallet::<T>::on_initialize(<frame_system::Pallet<T>>::block_number()); }
+	verify {
+		// Collators have been paid
+		for (col, initial) in collator_starting_balances {
+			assert!(T::Currency::free_balance(&col) > initial);
+		}
+		// Nominators have been paid
+		for (col, initial) in nominator_starting_balances {
+			assert!(T::Currency::free_balance(&col) > initial);
+		}
```

Comment:
The nominator balances changed as expected but for some odd reason the generated weight function does not depend on the number of nominations. It did before so it surprises me.

Reply:
fixed now